themes/default/content/blog/tour-pulumi-equinix-provider/index.md

```diff
@@ -0,0 +1,163 @@
+
+Now that your cluster is provisioned, you can deploy a workload (or any other Kubernetes or Helm resource) using the [Pulumi Kubernetes provider](https://www.pulumi.com/registry/packages/kubernetes/). In order to deploy Kubernetes resources to your cluster, you need to first declare an explicit provider. The explicit provider is needed because you are _creating the cluster and deploying workloads to that cluster in the same Pulumi program_. (For more details on default and explicit providers, see [Resource providers](https://www.pulumi.com/docs/concepts/resources/providers/).) Configure the provider to use the [Kubeconfig](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) of your newly created cluster:
+
+{{% chooser language "typescript,python" / %}}
+
+{{% choosable language typescript %}}
+
+```typescript
+const k8sProvider = new k8s.Provider("k8s-provider", {
+    kubeconfig: kubeconfig,
+});
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+k8s_provider = k8s.Provider(
+    "k8s-provider",
+    kubeconfig=kubeconfig
+)
+```
+
+{{% /choosable %}}
+
+You can then add resources to the cluster by specifying your explicit provider as a [resource option](https://www.pulumi.com/docs/concepts/options/):
+
+{{% chooser language "typescript,python" / %}}
+
+{{% choosable language typescript %}}
+
+```typescript
+const nginxDeployment = new k8s.apps.v1.Deployment("nginx-deployment", {
+    metadata: {
+        // ...
+    },
+    spec: {
+        // ...
+    },
+    }, { provider: k8sProvider });
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+deployment = k8s.apps.v1.Deployment(
+    "nginx-deployment",
+    metadata={
+        # ...
+    },
+    spec={
+        # ...
+    },
+    opts=pulumi.ResourceOptions(
+        provider=k8s_provider
+    )
+)
+```
+
+{{% /choosable %}}
+
+## Adding storage
+
+Our cluster as configured will not be able to run stateful workloads due to the lack of a [StorageClass](https://kubernetes.io/docs/concepts/storage/storage-classes/) from which [PersistentVolumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) can be provisioned. Popular storage options for bare-metal/edge Kubernetes clusters include [Portworx](https://www.purestorage.com/products/cloud-native-applications/portworx.html), [Longhorn](https://longhorn.io/), [Rook](https://rook.io/), or (for non-production scenarios as data loss is likely if a node goes down) [NFS](https://kubernetes.io/docs/concepts/storage/storage-classes/#nfs).
+
+## Learning more
+
+If you're interested in learning more about how to manage Equinix resources with Pulumi, check out [Pulumi's presentation at Equinix Demo Day 2023](https://youtu.be/-siv1ga0l_o) where we do a deeper dive into the basics of how Pulumi works and how to provision a Kubernetes cluster using Equinix Metal:
+
+{{< youtube "-siv1ga0l_o?feature=share&t=12576&rel=0" >}}
+
+{{% notes type="info" %}}
```

Comment:
I'd move this higher up in the post if possible.

Reply:
Done.